### PR TITLE
Improvements to docker image

### PR DESCRIPTION
**Pull request by duncan-brown**
This pull request simplifies the base PyCBC Docker container. It uses the standard `centos:centos7` container as a base and only installs the minimal set of software needed for a PyCBC install.

LALSuite is installed via `requirements.txt` so it the image will contain the latest release, since that is currently what is specified there. The Docker image is populated with a few ROM files, but not the complete set.

The appropriate files needed for OSG to turn this into a working singularity image are also included.

### CODE REVIEW

**Comment by ahnitz** [comment]
Adding Ian as a reviewer too, since this is probably something more than one person should understand how is going to work.

File: .travis.yml [hunk 1]
```diff
@@ -107,16 +107,15 @@ script:
     travis_retry docker exec -it pycbc_inst /bin/bash -lc 'df -h; exit $?' || exit 1 ;
     travis_retry docker exec -it pycbc_inst /bin/bash -lc 'cd /pycbc; pip install -r requirements.txt ; exit $?' || exit 1 ;
     travis_retry docker exec -it pycbc_inst /bin/bash -lc 'cd /pycbc; pip install . ; exit $?' || exit 1 ;
-    travis_retry docker exec -it pycbc_inst mkdir -p /opt/pycbc/src || exit 1;
-    travis_retry docker exec -it pycbc_inst /bin/bash -lc 'cd /opt/pycbc/src; git clone https://github.com/ligo-cbc/pycbc.git ; exit $?' || exit 1 ;
     if [ "x${TRAVIS_SECURE_ENV_VARS}" == "xtrue" ] ; then
-      LAL_EXTRA_TAG="e02dab8c" ;
+      LAL_EXTRA_TAG="current" ;
       travis_retry docker exec -it pycbc_inst mkdir -p /opt/pycbc/pycbc-software/share/lal-data || exit 1 ;
```
Comment (ahnitz) [inline .travis.yml]: Any way we can remove most of the travis commands (or all) from .travis.yml. Some of these at least probably should just be in the dockerfile to begin with such as the lal-data stuff, no?

Comment (duncan-brown) [reply .travis.yml]: @ahnitz I moved these to a script inside the Docker image. The way Travis works is that you need to mount the Travis pwd inside the container and install that.

**Comment by duncan-brown** [comment]
@ahnitz ok, I've made all the changes and I think this is working, but in order to fully test it needs to be merged to master, as only master has the privileges to try and push the build image to Dockerhub.

Please can you approve so I can merge and test this? Once it's merged, I'll open another pull request to fix any issues and fix the documentation once I've checked that the image can be used.

**Comment by spxiwh** [comment]
No issues from my side, so will let @ahnitz approve this for merging when he's happy. @duncan-brown how big is this Docker image going to be? I know that bundling the various `RUN` commands together into a single command used to make a big difference for file sizes if this is becoming a large file.

https://blog.codeship.com/reduce-docker-image-size/

Some `yum clean`s and/or the pip equivalent might also help.

Also why are we demanding a specific MPI distribution? Will this easily apply to different systems, that might have different MPI installed normally?

**Comment by duncan-brown** [comment]
@spxiwh image size is one of the things I want to test once it's pushed. I think I combined sufficient RUN commands together, but I can squeeze more if it's too big.

The version of MPI is for when this gets turned into a singularity container for use on XSEDE (Comet specifically). If you're running the docker container, then you won't be using MPI so it can be ignored. For other clusters, we'll need to maintain other images with different MPI versions, but I want to check that this one works first.

@ahnitz please can you approve so I can merge and test?

**Comment by ahnitz** [comment]
@duncan-brown I don't have any major issues. The folder layout is a bit too spread out though. Could we get all the docker related files into a single top-level folder? (subfolders are fine)

**Comment by duncan-brown** [comment]
@ahnitz good idea, I moved everything into a single folder.

**Comment by duncan-brown** [comment]
@ahnitz this is good to go. Please can you approve?

**Pull request merged**